sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpProducer.cs

```diff
@@ -311,7 +327,12 @@ internal class AmqpProducer : TransportProducer
             options.MaximumSizeInBytes ??= MaximumMessageSize;
             Argument.AssertInRange(options.MaximumSizeInBytes.Value, EventHubProducerClient.MinimumBatchSizeLimit, MaximumMessageSize.Value, nameof(options.MaximumSizeInBytes));
 
-            return new AmqpEventBatch(MessageConverter, options, ActiveFeatures);
+            // If idempotent publishing is enabled, a specialized transport batch is needed to
+            // support sequencing semantics during send operations.
+
+            return ((ActiveFeatures & TransportProducerFeatures.IdempotentPublishing) == 0)
```

Comment:
This is nitpicky. I was just wondering if this should be abstracted into `AmqpEventBatch.Create(MessageConverter, options, TransportProducerFeatures, ActiveFeatures)` and then return the base.

I was immediately wondering why we access ActiveFeatures here and then also passing it to the batch creation.

Reply:
Interesting thought.  I'll prototype that and see how it feels.  

The feature flags are intended to support a few things forthcoming that may need to be considered as part of measurement, conversion, or sending.   Currently, it's just the idempotent coarse-grained flag.

Reply:
I like it; centralizes the concern.   I'll push this up shortly.